### DISCUSSION TITLE
How to load tokenizer config?

### DISCUSSION THREAD

**Discussion by linonetwo** [withcatai/node-llama-cpp#203] [Q&A] [answered]
https://github.com/QwenLM/Qwen1.5/issues/304#issuecomment-2058155957

> They are also provided in tokenizer.json as the standard practice in transformers

Do we have an API to load this?

**Comment by giladgd** [comment, marked as answer]
There’s a `JinjaChatWrapper` you can use, and it’s automatically get used when the model includes a chat template.
You can run this command to chat with a model, and look for the chat wrapper metadata that’s printed before the chat starts:
```
npx -n node-llama-cpp chat <model file path>
```

You can also use `fileInfo` on a model instance to read metadata.

**Reply by linonetwo** [reply]
How about the tokenizer.json? I think it can solve wrong token in CJK language 
![图片](https://github.com/withcatai/node-llama-cpp/assets/3746270/1b68a0e7-86db-44f1-845b-e04575682dd8)

**Reply by linonetwo** [reply]
I used to not passing the template, and still get bad result. So the problem must cause by tokenlizer, do you have guide to load it @giladgd ?

**Reply by giladgd** [reply]
You can access all the tokenizer related metadata on a model by using `model.fileInfo.metadata.tokenizer`.
The `tokenizer.json` you're talking about is probably the [original file used in the HF format of the model](https://huggingface.co/Qwen/CodeQwen1.5-7B-Chat/blob/main/tokenizer.json).
`model.fileInfo.metadata.tokenizer` is a transformed version of that file adapted to `llama.cpp` and embedded into `.gguf` files

**Reply by linonetwo** [reply]
Thanks, so GGUF should have everything embedded. Then I have no idea why it performs so bad...
